Parser/Token.cs

```diff
@@ -38,9 +39,23 @@ public enum E {
    // Utility function used to echo an error to the console
    public void PrintError () {
       if (Kind != ERROR) throw new Exception ("PrintError called on a non-error token");
-      Console.ForegroundColor = ConsoleColor.Yellow;
-      Console.WriteLine ($"At line {Line}, column {Column} of {Source.FileName}: {Text}");
-      Console.ResetColor ();
+      var str = $"File: {Source.FileName}";
+      WriteLine (str);
+      WriteLine (new string ('\u2500', str.Length));
+      int end = Line + 2;
+      for (int i = Line - 2; i <= end; i++) {
+         if (i < 1 || i > Source.Lines.Length) continue;
+         WriteLine ($"{i,4}\u2502{Source.Lines[i - 1]}");
+         if (i != Line) continue;
+         var left = 4 + Column;
+         while (left >= WindowWidth) left -= WindowWidth;
+         ForegroundColor = ConsoleColor.Yellow;
+         CursorLeft = left;
+         WriteLine ('^');
+         CursorLeft = Math.Min (Math.Max (0, left - Text.Length / 2), WindowWidth - Text.Length);

```

Comment:
Instead of using **CursorLeft**, if you just pad the output with spaces, then it will render correctly even when output is redirected to a file.